diesel/src/pg/types/primitives.rs

```diff
@@ -1,14 +1,28 @@
 use std::io::prelude::*;
 
 use deserialize::{self, FromSql};
-use pg::Pg;
+use pg::{Pg, PgValue};
 use serialize::{self, IsNull, Output, ToSql};
 use sql_types;
 
+/// The returned pointer is *only* valid for the lifetime to the argument of
+/// `from_sql`. This impl is intended for uses where you want to write a new
+/// impl in terms of `String`, but don't want to allocate. We have to return a
+/// raw pointer instead of a reference with a lifetime due to the structure of
+/// `FromSql`
+impl FromSql<sql_types::Text, Pg> for *const str {
+    fn from_sql<'a>(bytes: Option<PgValue>) -> deserialize::Result<Self> {
```

Comment:
Remove the unused lifetime
```suggestion
    fn from_sql(bytes: Option<PgValue>) -> deserialize::Result<Self> {
```